### ISSUE TITLE
BAD-SAD for invalid mapping constructor

### ISSUE THREAD

**Issue by rdhananjaya** [ballerina-platform/ballerina-lang#29424]
**Description:**
```
function arrayToR(string[] ar) returns R {
    return {
        age: checkpanic int:fromString(ar[1]);
        id: ar[3]; 
    };
}
```

Output:
```
[Error - 1:30:28 PM] Operation 'text/codeAction' failed! {uri: '/Users/dhananjaya/Work/Samples/simple-etl/simpletl/main.bal', [10:1]- [15:2], error: 'class io.ballerina.compiler.syntax.tree.BasicLiteralNode cannot be cast to class io.ballerina.compiler.syntax.tree.BindingPatternNode (io.ballerina.compiler.syntax.tree.BasicLiteralNode and io.ballerina.compiler.syntax.tree.BindingPatternNode are in unnamed module of loader 'app')'}
java.lang.ClassCastException: class io.ballerina.compiler.syntax.tree.BasicLiteralNode cannot be cast to class io.ballerina.compiler.syntax.tree.BindingPatternNode (io.ballerina.compiler.syntax.tree.BasicLiteralNode and io.ballerina.compiler.syntax.tree.BindingPatternNode are in unnamed module of loader 'app')
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.getBLangVariableNode(BLangNodeTransformer.java:4717)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.createBLangVarDef(BLangNodeTransformer.java:2722)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:2696)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:472)
	at io.ballerina.compiler.syntax.tree.VariableDeclarationNode.apply(VariableDeclarationNode.java:67)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.generateAndAddBLangStatements(BLangNodeTransformer.java:4813)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:1470)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:472)
	at io.ballerina.compiler.syntax.tree.FunctionBodyBlockNode.apply(FunctionBodyBlockNode.java:59)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.populateFunctionNode(BLangNodeTransformer.java:1378)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.createFunctionNode(BLangNodeTransformer.java:1358)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:1336)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:472)
	at io.ballerina.compiler.syntax.tree.FunctionDefinitionNode.apply(FunctionDefinitionNode.java:71)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:589)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.transform(BLangNodeTransformer.java:472)
	at io.ballerina.compiler.syntax.tree.ModulePartNode.apply(ModulePartNode.java:54)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeTransformer.accept(BLangNodeTransformer.java:502)
	at io.ballerina.projects.DocumentContext.compilationUnit(DocumentContext.java:111)
	at io.ballerina.projects.ModuleContext.compileInternal(ModuleContext.java:357)
	at io.ballerina.projects.ModuleCompilationState$3.compile(ModuleCompilationState.java:100)
	at io.ballerina.projects.ModuleContext.compile(ModuleContext.java:323)
	at io.ballerina.projects.PackageCompilation.compileModulesInternal(PackageCompilation.java:168)
	at io.ballerina.projects.PackageCompilation.compileModules(PackageCompilation.java:160)
	at io.ballerina.projects.PackageCompilation.from(PackageCompilation.java:96)
	at io.ballerina.projects.PackageContext.getPackageCompilation(PackageContext.java:206)
	at io.ballerina.projects.Package.getCompilation(Package.java:130)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.waitAndGetPackageCompilation(BallerinaWorkspaceManager.java:217)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.semanticModel(BallerinaWorkspaceManager.java:196)
	at org.ballerinalang.langserver.codeaction.CodeActionRouter.getAvailableCodeActions(CodeActionRouter.java:65)
	at org.ballerinalang.langserver.codeaction.BallerinaCodeActionExtension.execute(BallerinaCodeActionExtension.java:46)
	at org.ballerinalang.langserver.codeaction.BallerinaCodeActionExtension.execute(BallerinaCodeActionExtension.java:34)
	at org.ballerinalang.langserver.LangExtensionDelegator.codeActions(LangExtensionDelegator.java:133)
	at org.ballerinalang.langserver.BallerinaTextDocumentService.lambda$codeAction$8(BallerinaTextDocumentService.java:335)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1692)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)

```

**Steps to reproduce:**

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

**Comment by rdhananjaya** [comment]
Seems to be fixed now.

**Comment by grainier** [comment]
Fixed in: #29231
